cmd/main.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"flag"
+
+	"com.ciandt.juggler/internal"
+)
+
+func main() {
+
+	var outbound = flag.String("outbound", "localhost:5000", "Outbound destination server")
```

Comment:
I think the idea for juggler was to change environments dynamically in order to have a single source of configuration, preferably without killing/restarting the process that's consuming those configs. There's two ways of doing this:

- Rewrite the running processes' environment variables on runtime (not recommended since there's no standard way of doing this without nasty hacks - we could also start a new shell with the new vars, but that can be hard to manage). 

- Use `juggler` as a "proxy" for connections. In that case we wouldn't  rewrite environment variables, but have a daemon that could switch which addresses they're proxying to via a command-line tool. I think that due to the multi-process, concurrent nature of Go that would be doable.

I think that can be done with raw TCP/UDP connections, and by dealing directly with byte streams. That could match your use case.

That's basically a long way to say that I'm a bit resistant to adding functionality for specific protocols for now (SOCKS5 in this case). Let's focus on the generic approach of forwarding TCP/UDP packets and if we have problems with that we can try to come up with specific functionality.

Reply:
It's mandatory to add a transport layer when proxying the TCP/UDP. If you don't do that the net package is going to add a default transport. What I'm doing here is manually specifying the Socks5 transport because the main environment is behind a firewall..

However, I think that we do can change the structure (currently the socks5 is an optional param) to include more transports layers and act more like a raw TCP forward.

Reply:
_It's mandatory to add a transport layer when proxying the TCP/UDP. If you don't do that the net package is going to add a default transport._

You mean you cannot create "raw" socket connections in Go? You need to actually specify a layer 5 protocol or it will default to something?